### ISSUE TITLE
[Spike] Investigate React Tree Table approach

### ISSUE THREAD

**Issue by mcarrano** [patternfly/patternfly-react#3141]
Determine the best path forward for introducing a tree table component. A design proposal can be found here: https://docs.google.com/document/d/1KXwK7rFXUBQ1RDn-VaxMepe3hoCb8Sl4i_4K_EaM-kg/edit?usp=sharing

Tree table is a complex component and it would be good to evaluate approaches to implementing in React.  Goals/outcomes for this spike:

* Implement basic tree-table based on the existing designs for Table, but with parent-child relationship between rows. Minimal requirements include:

* Support the ability to expand/collapse parent rows to show/hide children.

- Be responsive by supporting mobile as well as desktop views.

Filter and sorting capability is not required for the PoC, but it would be good to keep these in mind to be added in future iterations.

cc @LHinson @mturley @rachael-phillips

**Comment by mcarrano** [comment]
@rachael-phillips @mturley I have updated the description of this based on our discussion. Let me know if I missed anything.

**Comment by mturley** [comment]
Looks good to me, thanks @mcarrano!

**Comment by mturley** [comment]
Most of the discussion around this component is happening on a patternfly-next issue, I just want to drop this link here for context: https://github.com/patternfly/patternfly-next/issues/348

**Comment by mturley** [comment]
After some discussion with @mcarrano and @tlabaj, we decided this spike should be closed and instead we should open a spike on the core repo to investigate potential HTML+CSS structures for a responsive mobile-friendly tree table view.

I researched existing third-party libraries for tree views, and while some might have been usable if we only had desktop browsers in mind, I couldn't find any that seemed to meet our needs (theme-able to match PF4 and responsive enough to display multiple columns on desktop and collapse into a forward/back navigation scheme on mobile, as designed in https://github.com/patternfly/patternfly-next/issues/348). I then started working on a POC based on the PF4 Table HTML and CSS to see if that might be viable, but ran into some issues that convinced me we need to approach this from a core perspective and figure out the right HTML and CSS to use.

I'm closing this issue and I've opened the core spike issue here: https://github.com/patternfly/patternfly-next/issues/2455
